packages/helix-shared-storage/src/storage.js

```diff
@@ -465,8 +473,14 @@ class Bucket {
       };
       try {
         if (opts.addMetadata) {
-          const meta = await this.metadata(task.src) ?? {};
-          input.Metadata = { ...meta, ...opts.addMetadata };
+          const headers = await this.head(task.src);
+          if (headers) {
+            input.ContentType = headers.ContentType;
+            input.ContentEncoding = headers.ContentEncoding;
+            input.CacheControl = headers.CacheControl;
+            input.ContentDisposition = headers.ContentDisposition;
+          }
```

Comment:
This construct doesn't check for the input values to be something else than 'undefined'. 

What about using a list or property _names_ and then only set `input[name]` when `headers[name]` is set, e.g.:
```
['ContentType', 'ContentEncoding', 'CacheControl', 'ContentDisposition'].forEach((name) => {
  if (headers[name]) {
    input[name] = headers[name];
  }
}
```